qiskit_nature/mappers/second_quantization/logarithmic_mapper.py

```diff
@@ -0,0 +1,208 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Logarithmic Mapper."""
+
+import operator
+
+from fractions import Fraction
+from functools import reduce
+from typing import List, Union, Tuple
+
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import SparsePauliOp, Operator
+from qiskit_nature.operators.second_quantization import SpinOp
+from .spin_mapper import SpinMapper
+
+
+class LogarithmicMapper(SpinMapper):
+    """A mapper for Logarithmic spin-to-qubit mapping [1].
+    In this local encoding transformation, each individual spin S system is represented via
+    the lowest lying 2S+1 states in a qubit system with the minimal number of qubits needed to
+    represent >= 2S+1 distinct states.
+
+    [1]: Mathis, S. V., Guglielmo, M., & Ivano, T. (2020).
```

Comment:
```suggestion
    [1]: S. V. Mathis, G. Mazzola and I. Tavernelli (2020).
```
This makes the use of first- and surnames consistent. I am also still not sure about the `(2020)` at the end.. @pbark do you have a suggestion here?

Reply:
Thank you so much for reviewing the PR! I fixed everything. 

I'm not sure about the `(2020)` either. I just copied from this page, https://journals.aps.org/prd/abstract/10.1103/PhysRevD.102.094501. If we have something more appropriate, I have no problem to change it.  @mrossinek @pbark

Reply:
I thought about this some more but forgot to update my comment here. How about we leave the names on a separate line and append the `(2020).` at the end of last line which indices the journal, volume and pages.

Reply:
I think rather this:
```
Phys. Rev. D, 102 (9), 094501 (2020).
https://doi.org/10.1103/PhysRevD.102.094501
```

Reply:
Thx! I fixed it.